### PR TITLE
Renewal of `pymdp` documentation

### PR DESCRIPTION
**Pull request by conorheins**
### Summary

This addresses #199, which was targeted at a part of the documentation to aid NumPy-familiar users with migrating to `JAX`,  but also just generally renovates the full documentation suite.

This PR introduces a long-awaited overhaul of `pymdp`'s documentation, now created through [Mkdocs](https://www.mkdocs.org/) but still hosted by the same readthedocs URL.

The latest documentation introduced by this branch can be found here: https://pymdp-rtd.readthedocs.io/en/docs-rehaul/ 

In addition to a makeover of the documentation itself, the documentation most notably includes a new [NumPy-to-JAX migration guide](https://pymdp-rtd.readthedocs.io/en/docs-rehaul/migration/numpy-to-jax/) and a few new tutorial notebooks:
- a JAX-reimplementation of the [cue-chaining demo](https://pymdp-rtd.readthedocs.io/en/docs-rehaul/tutorials/notebooks/examples/envs/cue_chaining_demo/) (same one shown in the README.md of the master branch of pymdp)
- a more parameter-identifiable version of the [`fitting_with_pybefit.ipynb`](https://pymdp-rtd.readthedocs.io/en/docs-rehaul/tutorials/notebooks/examples/model_fitting/fitting_with_pybefit/) notebook.

### Suggested focus for reviewers 

Although GitHub shows ~21k changed lines, most of that is notebook JSON and mirrored notebook copies for docs publishing (in `pymdp/docs-mkdocs/tutorials/notebooks/`)

- ~14.7k lines are auto-synced notebook copies in the docs notebook tree
- Core `pymdp` module edits are largely typing/docstring expansion for API docs quality (majority of added lines in `pymdp/*.py`). 

Please prioritize reviewing **docs UX / readability**:

1. Navigation and discoverability (home, quickstart, migration, guides, notebook gallery).
2. Readability and clarity of new docs pages.
3. pymdp "gotchas" or hardpoints that deserve mention
4. Overall docs look/feel and consistency.

### Details
- Migrates docs infrastructure from Sphinx-style setup to MkDocs + MkDocstrings (`mkdocs.yml`), while keeping ReadTheDocs hosting.
- Adds docs CI (`.github/workflows/docs.yml`) and updates RTD config (`.readthedocs.yml`) for strict MkDocs builds.
- Adds docs utility scripts:
  - `scripts/docs_build.sh`
  - `scripts/docs_serve.sh`
  - `scripts/docs_sync_and_serve.sh`
  - `scripts/sync_docs_notebooks.sh`
  - Use `docs_sync_and_serve.sh` as a way to build notebooks/serve docs in one go (hosts on a local server).
- Introduces a manifest-driven curated notebook pipeline (`docs-mkdocs/tutorials/notebooks.manifest`) and synced notebook docs pages.
- Standardizes/expands docstrings and type annotations across core JAX modules (e.g. `pymdp/agent.py`, `pymdp/inference.py`, `pymdp/control.py`, `pymdp/learning.py`, `pymdp/maths.py`, `pymdp/utils.py`) to improve API reference quality.
- Updates README and installation/contributing guidance to an `uv`-first workflow.

### Drive-by fixes
This addresses #298 and #254 in the crossfire since it improves the parameter identifiability of the pybefit demo and re-implements the cue chaining demo in JAX.

#254 relevant changes
- Adds JAX cue-chaining environment + tutorial/demo assets:
  - `pymdp/envs/cue_chaining.py`
  - `examples/envs/cue_chaining_demo.ipynb`
  - `examples/envs/chained_cue_navigation.py`
  - updated README GIF assets

#298 relevant changes
- Adds pybefit/model-fitting recoverability workflow updates:
  - `examples/model_fitting/tmaze_recoverability.py`
  - updated `examples/model_fitting/fitting_with_pybefit.ipynb`

And relevant for both of them, it adds regression/smoke tests for new tutorial-facing functionality:
  - `test/test_cue_chaining_env.py`
  - `test/test_pybefit_model_fitting.py`
  - `test/test_tmaze_recoverability.py`

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -36,7 +36,7 @@ dependencies = [
 
 [dependency-groups]
 test = [
-    "ipykernel>=7.1.0",
+    "ipykernel>=6.29.0,<7",
```
[the remark this answers is not in the document]
Comment (conorheins) [reply pyproject.toml:40]: This version is needed in order to run [`mkdocs-jupyter`](https://pypi.org/project/mkdocs-jupyter/), that's why we have to restrict it. I added an inline note in `pyproject.toml` documenting this in 4a75aba, and we can relax the bound once `mkdocs-jupyter` supports ipykernel 7.

File: pyproject.toml [hunk 2]
```diff
@@ -47,12 +47,26 @@ test = [
...
     'jax[cuda12]>=0.3.4',
     'jaxlib[cuda12]>=0.3.4',
 ]
+docs = [
+    "mkdocs>=1.6.1",
+    "mkdocs-material>=9.5.50",
+    "mkdocstrings[python]>=0.27.0",
+    "mkdocs-jupyter>=0.24.8",
+    "mkdocs-redirects>=1.2.2",
+]
```
[the remark this answers is not in the document]
Comment (conorheins) [reply pyproject.toml]: Removed the extra from `[dependency-groups]` in commit 4a75aba

**Review by riddhipits — COMMENTED** [review]
Hey Conor - amazing work on this! 

As requested in your email, here is the high- and low- level feedback. 

High-level feedback: 
1. I'm confused about why the canonical usage of rollout() in the [rollout guide](https://pymdp-rtd.readthedocs.io/en/docs-rehaul/guides/rollout-active-inference-loop/) says we need to jit the rollout as rollout() uses jax.scan internally, ensuring it is jax-traced and compiled when first used. So we shouldn't need to jit it again, right?
2. In the batching and batch size section of the [migration document](https://pymdp-rtd.readthedocs.io/en/docs-rehaul/migration/numpy-to-jax/), I think we should mention what the defaults are - if I remember correctly, the environment is unbatched but Agent is, assuming the layout of 1 env and multiple agents in that 1 env. 
3. Do we need docs/ now that docs-mkdocs/ exists
4. Do we want to create a API reference sheet for si planning algo?

Low-level feedback: 
1. I took my friend's computer who didn't even have command line tools installed and I followed the installation guide it worked perfectly. 
2. The bullet points [in this page](https://pymdp-rtd.readthedocs.io/en/docs-rehaul/migration/numpy-to-jax/) beginning "For environment-side batching, either:" is not correctly formatted
3. Typo in [distribution api tutorial](https://pymdp-rtd.readthedocs.io/en/docs-rehaul/tutorials/notebooks/examples/api/model_construction_tutorial/) "gamma = 10 # deterministic behavior; make it gamma smaller for stochastic behavior" with the "make it" 
4. I'm unsure what the difference is between "Tutorials", "Guides", and "Notebooks". In the [main documentation page](https://pymdp-rtd.readthedocs.io/en/docs-rehaul/), there's a section saying "Tutorials and notebooks", with links to Tutorial overview and Notebook gallery. But the Tutorial overview page, there's the Notebook gallery linked along with two guides. Might be cleaner to just have the Notebook gallery linked in the list under "Start here" and remove the tutorials and notebook section entirely as the guides are already linked in the start-here list. 
5. I am unsure what [this api overview page](https://pymdp-rtd.readthedocs.io/en/docs-rehaul/api/ ) is for. 
6. While I think it's clear to us what the distribution class does, I don't think the name of the feature is self-explanatory so I think we need to rename it to make it more self-explanatory and clear especially given how important and useful of a feature it is for new users. Re: referring to it as "Model and Distribution Tutorial" in [here](https://pymdp-rtd.readthedocs.io/en/docs-rehaul/tutorials/notebooks/)
7. In [generative model structure doc](https://pymdp-rtd.readthedocs.io/en/docs-rehaul/guides/generative-model-structure/), it should be A_dependencies[m] not A_dependencies[f].
8. Duplicate "this" in "see also this [this post](https://builtin.com/data-science/numpy-random-seed)" in [this page](https://pymdp-rtd.readthedocs.io/en/docs-rehaul/migration/numpy-to-jax/)
9. in the readme, docs badge links to docs-rehaul - after merge into v1.0.0_alpha (and later master), we may need to update the badge accordingly
